Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -1971,23 +1955,136 @@ void UCesiumGltfComponent::FinishDestroy() {
   Super::FinishDestroy();
 }
 
-#if !PHYSICS_INTERFACE_PHYSX
-// This is copied from FChaosDerivedDataCooker::BuildTriangleMeshes in
-// C:\Program Files\Epic
-// Games\UE_4.26\Engine\Source\Runtime\Engine\Private\PhysicsEngine\Experimental\ChaosDerivedData.cpp.
-// We can't use that method directly because it is private. Since we've copied
-// it anyway, we've also modified it to be more efficient for our particular
-// input data.
+#if PHYSICS_INTERFACE_PHYSX
+static void BuildPhysXTriangleMeshes(
+    PxTriangleMesh*& pCollisionMesh,
+    const IPhysXCooking* pPhysXCooking,
+    const TArray<FStaticMeshBuildVertex>& vertexData,
+    const TArray<uint32>& indices) {
+
+  if (pPhysXCooking) {
+    int32 vertexCount = vertexData.Num();
+    int32 triangleCount = indices.Num() / 3;
+
+    // TODO: use PhysX interface directly so we don't need to copy the
+    // vertices (it takes a stride parameter).
+    TArray<FVector> vertices;
+    vertices.SetNum(vertexCount);
+
+    for (size_t i = 0; i < vertexCount; ++i) {
+      vertices[i] = vertexData[i].Position;
+    }
+
+    TArray<FTriIndices> physicsIndices;
+    physicsIndices.SetNum(triangleCount);
+
+    for (size_t i = 0; i < triangleCount; ++i) {
+      physicsIndices[i].v0 = indices[3 * i];
+      physicsIndices[i].v1 = indices[3 * i + 1];
+      physicsIndices[i].v2 = indices[3 * i + 2];
+    }
+
+    pPhysXCooking->CreateTriMesh(
+        "PhysXGeneric",
+        EPhysXMeshCookFlags::Default,
+        vertices,
+        physicsIndices,
+        TArray<uint16>(),
+        true,
+        pCollisionMesh);
+  }
+}
+
+#else
+template <typename TIndex>
+static bool isDegenerate(
+    const Chaos::TVector<TIndex, 3>& triangle,
+    const TArray<FStaticMeshBuildVertex>& vertexData) {
```

Comment:
I don't think this check is actually required, this came from trying to build meshes similar to how Unreal does it, but in our case we can be somewhat confident going in that our triangles are not ill-formed after the tiling process. Would you agree @kring?